ksql-udf/src/main/java/io/confluent/ksql/function/udf/Udf.java

```diff
@@ -37,4 +37,13 @@
    * @return the text to display to the user.
    */
   String description() default "";
+
+  /**
+   * The schema for the return value of the UDF.
+   *
+   * <p>For simple method signatures, this is optional and can be determined from
+   * the return value itself. For complex return types (e.g. {@code Struct} types),
+   * this is required and will fail if not supplied.
```

Comment:
What does this failure look like? Will it prevent the KSQL server from starting? What's the error message?

Reply:
It fails while the server is starting up with `Type is not supported: STRUCT`. I changed this to `Type inference is not supported for type: STRUCT`

Reply:
```

Reply:
Works for me. Ideally the error message would contain a note about specifying the schema in the UDF annotation, but if getting that custom message in is too much of a hack, I'm fine leaving it as is. Thanks for explaining!